### PR TITLE
Adjust logic and add unit test

### PR DESCRIPTION
**Pull request by rajivnathan**
Unit tests for https://github.com/codeready-toolchain/member-operator/pull/337

After adding the unit test with a User/Identity that had nil labels the tests were failing because the assertUser function checks the following:
```
assert.Equal(t, userAcc.Name, user.Labels["toolchain.dev.openshift.com/owner"])
assert.Equal(t, toolchainv1alpha1.ProviderLabelValue, user.Labels[toolchainv1alpha1.ProviderLabelKey])
```

So I'm assuming it's always expected that both owner and provider labels should be present? At least that's the assumption I went with to make the unit tests pass. I changed the `addProviderLabel` func to `addOwnerAndProviderLabels` so it adds both. Please verify that that's what we actually want.

### CODE REVIEW

File: controllers/useraccount/useraccount_controller.go [hunk 1]
```diff
@@ -385,13 +385,29 @@ func setOwnerAndProviderLabel(object metav1.Object, owner string) {
...
 		labels[toolchainv1alpha1.ProviderLabelKey] = toolchainv1alpha1.ProviderLabelValue
+		object.SetLabels(labels)
+		changed = true
+	}
+
+	if _, exists := object.GetLabels()[toolchainv1alpha1.OwnerLabelKey]; !exists {
+		labels := object.GetLabels()
+		if labels == nil {
+			labels = make(map[string]string)
+		}
```
Comment (MatousJobanek) [inline controllers/useraccount/useraccount_controller.go:404]: it's just a detail, but this case shouldn't ever happen - if the labels are nil, then they would be set in the first case (when checking provider label)

**Review by MatousJobanek — APPROVED** [review]
